components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -282,4 +291,65 @@ private void removeEntriesFromCache(ServiceProvider serviceProvider, String tena
+     * Stores the value of SaaS property before application is updated.
+     *
+     * @param serviceProvider Service Provider
+     * @param tenantDomain    Application tenant domain
+     * @throws IdentityApplicationManagementException
+     */
+    private void storeSaaSPropertyValue(ServiceProvider serviceProvider, String tenantDomain) throws IdentityApplicationManagementException {
+
+        ServiceProvider sp = OAuth2ServiceComponentHolder.getApplicationMgtService()
+                .getServiceProvider(serviceProvider.getApplicationName(), tenantDomain);
+        IdentityUtil.threadLocalProperties.get().put(SAAS_PROPERTY, sp.isSaasApp());
+    }
+
+    /**
+     * Revokes access tokens of OAuth applications if SaaS is disabled.
+     *
+     * @param serviceProvider Service Provider
+     * @param tenantDomain    Application tenant domain
+     * @throws IdentityApplicationManagementException
+     */
+    private void revokeAccessTokensWhenSaaSDisabled(final ServiceProvider serviceProvider, final String tenantDomain) throws IdentityApplicationManagementException {
+
+        try {
+            boolean wasSaasEnaledBefore = false;
+            Object saasStatus = IdentityUtil.threadLocalProperties.get().get(SAAS_PROPERTY);
+            if (saasStatus instanceof Boolean) {
+                wasSaasEnaledBefore = (Boolean) saasStatus;
+            }
+            if (wasSaasEnaledBefore && !serviceProvider.isSaasApp()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("SaaS setting removed for application: " + serviceProvider.getApplicationName()
+                            + "in tenant domain: " + tenantDomain + ", hence proceeding to token revocation of other tenants.");
+                }
+                final int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+                final TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
+
+                new Thread(new Runnable() {
+                    public void run() {
+                        InboundAuthenticationRequestConfig[] configs = serviceProvider.getInboundAuthenticationConfig()
+                                .getInboundAuthenticationRequestConfigs();
+                        for (InboundAuthenticationRequestConfig config : configs) {
+                            if (IdentityApplicationConstants.OAuth2.NAME.equalsIgnoreCase(config.getInboundAuthType()) &&
+                                    config.getInboundAuthKey() != null) {
+                                String oauthKey = config.getInboundAuthKey();
+                                try {
+                                    tokenMgtDAO.revokeSaaSTokensOfOtherTenants(oauthKey, tenantId);
+                                } catch (IdentityOAuth2Exception e) {
+                                    log.error("Error occurred while revoking access tokens for client ID: "
+                                            + config.getInboundAuthKey() + "and tenant domain: " + tenantDomain, e);
```

Comment:
"and tenant domain: " --> " and tenant domain: "
(leave a space)

Reply:
Fixed with #781

Reply:
add a whitespace before `and`